org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/WorkspaceManager.xtend

```diff
@@ -163,13 +163,16 @@ import org.eclipse.xtext.workspace.IWorkspaceConfig
     }
     
     protected def boolean exists(URI uri) {
-        getProjectManager(uri).resourceSet.URIConverter.exists(uri, null)
+        getProjectManager(uri)?.resourceSet.URIConverter.exists(uri, null)
     }
     
     def <T> T doRead(URI uri, (Document, XtextResource)=>T work) {
     	val resourceURI = uri.trimFragment
     	val projectMnr = getProjectManager(resourceURI)
-        val resource = projectMnr.getResource(resourceURI) as XtextResource
+        val resource = projectMnr?.getResource(resourceURI) as XtextResource
+        if (resource === null) {
+        	return work.apply(null, null)	
```

Comment:
It's ok for me, as it is now. If the resource is null, the document is, too. The situation also is not swallowed.